pkg/accumulator/accumulator.go

```diff
@@ -0,0 +1,256 @@
+package accumulator
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"sniffer/pkg/config"
+	"sniffer/pkg/ebpfeng"
+	evData "sniffer/pkg/ebpfev/v1"
+
+	logger "github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+)
+
+const (
+	ACCUMULATOR_SIZE = 10
+)
+
+const (
+	DROP_EVENT_OCCURRED = "drop event occurred\n"
+)
+
+type containersAccumulator struct {
+	accumulatorDataPerContainer map[string]chan evData.EventData
+	registerMutex               sync.Mutex
+}
+
+type Accumulator struct {
+	accumulatorData                 []map[string][]evData.EventData
+	syncReaderWriterAccumulatorData sync.Mutex
+	firstMapKeysOfAccumulatorData   []string
+	cacheAccumulatorSize            int
+	mainDataChannel                 chan *evData.EventData
+	containersData                  containersAccumulator
+	ebpfEngine                      ebpfeng.EbpfEngineClient
+}
+
+type ContainerAccumulator struct {
+	dataChannel chan evData.EventData
+	containerID string
+}
+
+var myContainerID string
+var accumulatorSingelton *Accumulator
```

Comment:
typo.. should be `Singleton`. 
Anyways, it can be simplified by calling it with some shorter name like `instance` (it's pretty obvious that it's a singleton)